File: examples/remix.js
// Pentagon Games Image Remix API — Node.js example (async: submit, then poll). Node 18+.
//   IMGEN_KEY=ef_xxx node remix.js my_pet.png "make it a knight in golden armor"
import { readFile, writeFile } from "node:fs/promises";

const BASE = process.env.IMGEN_BASE || "https://imgen.pentagon.games";
const KEY  = process.env.IMGEN_KEY;
const sleep = (ms) => new Promise((r) => setTimeout(r, ms));

async function remix(imagePath, prompt, out = "derivative.png") {
  const form = new FormData();
  form.set("image", new Blob([await readFile(imagePath)]), imagePath.split(/[\\/]/).pop());
  form.set("prompt", prompt + ", keep it a creature with its animal head and body, not human");
  form.set("negative_prompt", "blurry, low quality, deformed, human face, human person");
  form.set("steps", "35");

  // 1) submit
  const sub = await fetch(`${BASE}/v1/edit`, { method: "POST", headers: { "x-api-key": KEY }, body: form });
  if (!sub.ok) throw new Error(`submit ${sub.status} ${await sub.text()}`);
  const { job_id } = await sub.json();
  console.log("job", job_id);

  // 2) poll until 200 (202 = still working)
  for (;;) {
    const r = await fetch(`${BASE}/v1/result/${job_id}`, { headers: { "x-api-key": KEY } });
    if (r.status === 200) { await writeFile(out, Buffer.from(await r.arrayBuffer())); console.log("saved", out); return; }
    if (r.status === 202) { await sleep(10000); continue; }
    throw new Error(`result ${r.status} ${await r.text()}`);
  }
}

const [imagePath, prompt] = process.argv.slice(2);
remix(imagePath, prompt).catch((e) => { console.error(e); process.exit(1); });

File: examples/remix.py
#!/usr/bin/env python3
"""Pentagon Games Image Remix API — Python example (async: submit a job, poll for the result).

    pip install requests
    IMGEN_KEY=ef_xxx python remix.py my_pet.png "make it a knight in golden armor"
"""
import os, sys, time, requests

BASE = os.environ.get("IMGEN_BASE", "https://imgen.pentagon.games")
KEY  = os.environ["IMGEN_KEY"]          # export IMGEN_KEY=ef_...
H    = {"x-api-key": KEY}

def remix(image_path, prompt, out="derivative.png", poll=10, **params):
    # 1) submit the job
    with open(image_path, "rb") as f:
        r = requests.post(f"{BASE}/v1/edit", headers=H,
                          files={"image": f}, data={"prompt": prompt, **params})
    r.raise_for_status()
    job = r.json()["job_id"]
    print("job", job)
    # 2) poll until the PNG is ready (202 = still working)
    while True:
        rr = requests.get(f"{BASE}/v1/result/{job}", headers=H)
        if rr.status_code == 200:
            with open(out, "wb") as g:
                g.write(rr.content)
            print("saved", out); return out
        if rr.status_code == 202:
            print("...", rr.json().get("status")); time.sleep(poll); continue
        rr.raise_for_status()   # 404/500

if __name__ == "__main__":
    img, prompt = sys.argv[1], sys.argv[2]
    # keep_creature guardrail baked into the example
    remix(img, prompt + ", keep it a creature with its animal head and body, not human",
          negative_prompt="blurry, low quality, deformed, human face, human person",
          steps=35, cfg=4.0, seed=7)
